source/common/http/conn_manager_impl.h

```diff
@@ -293,8 +328,14 @@ class ConnectionManagerImpl : Logger::Loggable<Logger::Id::http>,
     void addStreamDecoderFilterWorker(StreamDecoderFilterSharedPtr filter, bool dual_filter);
     void addStreamEncoderFilterWorker(StreamEncoderFilterSharedPtr filter, bool dual_filter);
     void chargeStats(const HeaderMap& headers);
+    // Returns the encoder filter to start iteration with. If the function is called from a filter
+    // that should always iterate from the next filter, sets always_start_next to be true.
```

Comment:
, sets always_start_next to be true. -> always_start_next should be set to true

here and below.

Reply:
I'm also a bit confused why we need this.  given we shouldn't be able to set iterate_from_current_filter_ to true until after we've [en/de]coded 100 continue and headers, at which point I'd think we couldn't set it again.  Can you walk me through when this happens?

Reply:
In the latest version of commonContinue(), I set iterate_from_current_filter_ before doHeaders() is called (because this way, we can reset iteration state at the beginning). That's why iterate_from_current_filter_ can be true before [en/de]codeHeaders() (triggered by doHeaders()) is called.

I could save the current iteration state at the beginning of commonContinue(), and set iterate_from_current_filter_ after doHeaders(). This way, we don't need always_start_next. If this way is preferred, I will make the change. Thanks for the review!